cmd/kubeadm/app/discovery/file/file.go

```diff
@@ -55,13 +55,23 @@ func ValidateConfigInfo(config *clientcmdapi.Config, clustername string) (*clien
 	// This is the cluster object we've got from the cluster-info kubeconfig file
 	defaultCluster := kubeconfigutil.GetClusterFromKubeConfig(config)
 
+	var caCerts []byte
+	if defaultCluster.CertificateAuthority != "" {
+		caCerts, err = ioutil.ReadFile(defaultCluster.CertificateAuthority)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to load CA certificate referenced by kubeconfig")
```

Comment:
Maybe dumping `defaultCluster.CertificateAuthority` as part of the error message is good thing to do.